src/Java.Interop.Tools.JavaTypeSystem/Adapters/JavaXmlApiExporter.cs

```diff
@@ -0,0 +1,359 @@
+						WriteFullEndElement (writer);
+					}
+
+					WriteFullEndElement (writer);
+				}
+
+				WriteFullEndElement (writer);
+			}
+
+			WriteFullEndElement (writer);
+		}
+
+		static void SaveConstructor (JavaConstructorModel ctor, XmlWriter writer)
+			=> SaveMember (ctor, writer, "constructor", null, null, null, null, null, ctor.ParentType.FullName, null, null, null, ctor.Parameters, ctor.IsBridge, null, ctor.IsSynthetic, null);
+
+		static void SaveField (JavaFieldModel field, XmlWriter writer)
+		{
+			var value = field.Value;
+
+			if (value != null && (field.Type == "double" || field.Type == "float"))
+				value = value.Replace ("E+", "E");
+
+			SaveMember (field, writer, "field", null, null, null, null,
+				XmlConvert.ToString (field.IsTransient),
+				field.Type,
+				field.TypeGeneric,
+				value,
+				XmlConvert.ToString (field.IsVolatile),
+				null,
+				null,
+				null,
+				null,
+				field.IsNotNull);
+		}
+
+		static void SaveMethod (JavaMethodModel method, XmlWriter writer)
+		{
+			bool check (JavaMethodModel _) => _.BaseMethod?.ParentType?.Visibility == "public" &&
+			     !method.IsStatic &&
+					      method.Parameters.All (p => p.InstantiatedGenericArgumentName == null);
+
+			//// skip synthetic methods, that's what jar2xml does.
+			//// However, jar2xml is based on Java reflection and it generates synthetic methods
+			//// that actually needs to be generated in the output XML (they are not marked as
+			//// "synthetic" either by asm or java reflection), when:
+			//// - the synthetic method is actually from non-public ancestor class
+			////   (e.g. FileBackupHelperBase.writeNewStateDescription())
+			//// For such case, it does not skip generation.
+			if (method.IsSynthetic && (method.BaseMethod == null || check (method)))
+				return;
+
+			//// Here we skip most of the overriding methods of a virtual method, unless
+			//// - the method visibility or final-ity has changed: protected Object#clone() is often
+			////   overriden as public. In that case, we need a "new" method.
+			//// - the method is covariant. In that case we need another overload.
+			//// - they differ in "abstract" or "final" method attribute.
+			//// - the derived method is static.
+			//// - the base method is in the NON-public class.
+			//// - none of the arguments are type parameters.
+			//// - finally, it is the synthetic method already checked above.
+			if (method.BaseMethod != null &&
+			    !method.BaseMethod.IsAbstract &&
+			    method.BaseMethod.Visibility == method.Visibility &&
+			    method.BaseMethod.IsAbstract == method.IsAbstract &&
+			    method.BaseMethod.IsFinal == method.IsFinal &&
+			    !method.IsSynthetic &&
+			    check (method))
+				return;
+
+			SaveMember (method, writer, "method",
+				XmlConvert.ToString (method.IsAbstract),
+				XmlConvert.ToString (method.IsNative),
+				GetVisibleReturnTypeString (method),
+				XmlConvert.ToString (method.IsSynchronized),
+				null,
```

Comment:
At this point, could you please use named parameters, possibly for all of them?  This is *a lot* of parameters…